### PR TITLE
[Term Entry] Dart List.asMap()

### LINKED ISSUE #4298: [Term Entry] Dart List .asMap()
### Reason/inspiration (optional)

We would like a new term entry under Dart [List](https://www.codecademy.com/resources/docs/dart/concept/list) for the Dart List `.asMap()` method. 
Provide the entry in a markdown file under `docs/content/dart/concept/list/terms/asMap/asMap.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by dancikmad**
### Description

- Created a new term entry under Dart List **.asMap()** method

### Issue Solved

- Closes #4298 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/dart/concepts/list/terms/asMap/asMap.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+In Dart, the **`.asMap()`** method is used to obtain a new **`Map`** view of a **`List`**, where the keys are the indices of the list and the values are the elements at those indices. This method is particularly useful when you need to work with list elements in a map-line manner, accessing them by their indices.
+
+## Syntax
+
+```pseudo
+list.asMap()
+```
+
+- this method does not take any arguments.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/asMap/asMap.md:24]:
```suggestion
```
No need for this statement.

File: content/dart/concepts/list/terms/asMap/asMap.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+---
+Title: '.asMap()'
+Description: 'Returns a new **`Map`** view of a **`List`** where keys are the indices and values are elements'
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/asMap/asMap.md:3]:
```suggestion
Description: 'Returns an unmodifiable Map view of a List where keys are the indices and values are elements.'
```

File: content/dart/concepts/list/terms/asMap/asMap.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+Here is the output for the above code example:
+
+```shell
+Original List: [Steve, Tonny, Jack, Jamie]
+Map view of the list: {0: Steve, 1: Tonny, 2: Jack, 3: Jamie}
+```
+
+With this approach, the list undergoes converstion into a map. The resulting map remains immutable.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/asMap/asMap.md:48]: Add this as a note that the resulting map remains immutable.

File: content/dart/concepts/list/terms/asMap/asMap.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Tags:
+  - 'Methods'
+  - 'Dart'
+  - 'Lists'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+In Dart, the **`.asMap()`** method is used to obtain a new **`Map`** view of a **`List`**, where the keys are the indices of the list and the values are the elements at those indices. This method is particularly useful when you need to work with list elements in a map-line manner, accessing them by their indices.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/asMap/asMap.md:16]:
```suggestion
In Dart, the **`.asMap()`** method is used to obtain a new `Map` view of a `List`, where the keys are the indices of the list and the values are the elements at those indices. This method is particularly useful when you need to work with list elements in a map-line manner, accessing them by their indices.
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@dancikmad I have suggested some changes, can you make them?

**Comment by dancikmad** [comment]
Hi, @avdhoottt thank you for pointing out. I will start working on these changes.

**Comment by dancikmad** [comment]
@avdhoottt hi, i made changes after you review suggestions. let me know if it is all good now :)

File: content/dart/concepts/list/terms/asMap/asMap.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+```shell
+Original List: [Steve, Tonny, Jack, Jamie]
+Map view of the list: {0: Steve, 1: Tonny, 2: Jack, 3: Jamie}
+```
+
+With this approach, the list undergoes converstion into a map.
+
+> **Note**
+> The resulting map remains immutable.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/asMap/asMap.md:49]:
```suggestion
> **Note:** The resulting map remains immutable.
```

**Review by avdhoottt — APPROVED** [review]
Looks good for the second review.

**Pull request merged**